filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobFileSystemManager.scala

```diff
@@ -87,6 +102,44 @@ case class BlobFileSystemManager(
 
 sealed trait BlobSasTokenGenerator { def generateBlobSasToken: Try[AzureSasCredential] }
 object BlobSasTokenGenerator {
+
+  /**
+    * Creates the correct BlobSasTokenGenerator based on config inputs. This generator is responsible for producing
+    * a valid SAS token for use in accessing a Azure blob storage container.
+    * Two types of generators can be produced here:
+    * > Workspace Manager (WSM) mediated SAS token generator, used to create SAS tokens that allow access for
+    * blob containers mediated by the WSM, and is enabled when a WSM config is provided. This is what is intended
+    * for use inside Terra
+    *    OR
+    * > Native SAS token generator, which obtains a valid SAS token from your local environment to reach blob containers
+    * your local azure identity has access to and is the default if a WSM config is not found. This is intended for
+    * use outside of Terra
+    *
+    * Both of these generators require an authentication token to authorize the generation of the SAS token.
+    * See BlobSasTokenGenerator for more information on how these generators work.
+    *
+    * @param config A BlobFileSystemConfig object
+    * @return An appropriate BlobSasTokenGenerator
+    */
+  def createBlobTokenGeneratorFromConfig(config: BlobFileSystemConfig): BlobSasTokenGenerator =
+    config.workspaceManagerConfig.map { wsmConfig =>
+      val wsmClient: WorkspaceManagerApiClientProvider = new HttpWorkspaceManagerClientProvider(wsmConfig.url)
+
+      // WSM-mediated mediated SAS token generator
+      // parameterizing client instead of URL to make injecting mock client possible
+      BlobSasTokenGenerator.createBlobTokenGenerator(
+        config.blobContainerName,
+        config.endpointURL,
+        wsmConfig.workspaceId,
+        wsmConfig.containerResourceId,
+        wsmClient,
+        wsmConfig.overrideWsmAuthToken
+      )
+    }.getOrElse(
+      // Native SAS token generator
+      BlobSasTokenGenerator.createBlobTokenGenerator(config.blobContainerName, config.endpointURL, config.subscriptionId)
+    )
```

Comment:
How seriously do we take these code coverage warnings?

Reply:
Not very. You can remove them by going to the three dots in the upper left and unchecking Show Annotations.

We don't usually let these or the GHA code coverage warnings block merging, but it's a signal we should be aware of. A trigger to consider - is it okay that this line isn't tested?